src/wallet/signer.ts

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable import/max-dependencies -- These imports are required for Signer */
+import { BigNumber } from "bignumber.js";
+import { flatMap } from "lodash";
+import { decodeAccountID } from "ripple-address-codec";
+import {
+  decode,
+  encodeForSigning,
+  encodeForSigningClaim,
+} from "ripple-binary-codec";
+import {
+  sign as signWithKeypair,
+  verify as verifySignature,
+} from "ripple-keypairs";
+
+import { ValidationError } from "../common/errors";
+import { Signer } from "../models/common";
+import { Transaction } from "../models/transactions";
+import { verifyBaseTransaction } from "../models/transactions/common";
+import Wallet from "../Wallet";
+
+/**
+ * Uses a wallet to cryptographically sign a transaction which proves the owner of the wallet
+ * is issuing this transaction.
+ *
+ * @param wallet - A Wallet that holds your cryptographic keys.
+ * @param tx - The Transaction that is being signed.
+ * @param multisign - If true, changes the signature format to encode for multisigning.
+ * @returns A signed Transaction.
+ */
+function sign(
+  wallet: Wallet,
+  tx: Transaction,
+  multisign: boolean = false
+): string {
+  return wallet.signTransaction(
+    tx,
+    multisign ? { signAs: wallet.getClassicAddress() } : { signAs: "" }
+  );
+}
+
+/**
+ * Takes several transactions (in object or blob form) and creates a single transaction with all Signers
+ * that then gets signed and returned.
+ *
+ * @param transactions - An array of Transactions (in object or blob form) to combine and sign.
+ * @returns A single signed Transaction which has all Signers from transactions within it.
+ * @throws ValidationError if:
+ * - There were no transactions given to sign
+ * - The SigningPubKey field is not the empty string in any given transaction
+ * - Any transaction is missing a Signers field.
+ */
+function combineMultisigned(
```

Comment:
Do you think `combineMultisigned` is a better name than just `multisign`?

Reply:
Now that we've moved the signing prep into `sign` I think `multisign` is a better name than `combineMultisigned` (Because it's no longer ambiguous which part of the process it's referring to)